evaluator/evaluator.go

```diff
@@ -258,6 +259,8 @@ func evalFloatInfixExpression(operator string, left object.Object, right object.
 		return nativeBoolToBooleanObject(lVal == rVal)
 	case "!=":
 		return nativeBoolToBooleanObject(lVal != rVal)
+	case "%":
```

Comment:
I hit a strange error where dividing a number turns it into a Float which is no longer valid for modulus. I just decided to add modulus for floats but lmk if you think this is a good idea.

Reply:
Related: I think, the error system is also a little broken right now I think (errors are not getting piped up) but I think that should be a separate pr

Reply:
This is awesome, and thanks for making the other PR too!

Totally down to have modulus for floats too! Another way we could do this is cast to an int above, so `collatz(int(n/2))` _or_ `if (int(n) % 2 == 0)`

Reply:
oh did not know there were casts! I feel like float modulus is useful(?) so might merge that part in